src/connector/connector.ts

```diff
@@ -1,10 +1,28 @@
 import { Channel, PresenceChannel } from './../channel';
 
+type CustomConnector = new (options: Options) => Connector;
+
+type Broadcaster = 'socket.io' | 'pusher' | 'null' | CustomConnector
+
+export interface Options {
+    auth?: {
+        headers: Record<string, string>
+    },
+    userAuthentication?: {
```

Comment:
Haven't set up fields that are isn't actually referenced in this package. For service specific options the user can import typing from the service's package and use an intersection of types.

Reply:
In fact `authEndpoint` for example is deprecated in pusher.